sgx/types.go

```diff
@@ -0,0 +1,38 @@
+package sgx
+
+import (
+	"github.com/edgelesssys/ego/enclave"
+	log "github.com/sirupsen/logrus"
+)
+
+type EnclaveInfo struct {
+	ProductID []byte
+	SignerID  []byte
+	UniqueID  []byte
+}
+
+func NewEnclaveInfo(productID, signerID, uniqueID []byte) *EnclaveInfo {
+	return &EnclaveInfo{
+		ProductID: productID,
+		SignerID:  signerID,
+		UniqueID:  uniqueID,
+	}
+}
+
+// GetSelfEnclaveInfo sets EnclaveInfo from self-generated remote report
+func GetSelfEnclaveInfo() (*EnclaveInfo, error) {
+	// generate self-remote-report and get product ID, signer ID, and unique ID
+	reportBz, err := GenerateRemoteReport([]byte(""))
+	if err != nil {
+		log.Errorf("failed to generate self-report: %v", err)
+		return nil, err
+	}
+
+	report, err := enclave.VerifyRemoteReport(reportBz)
+	if err != nil {
+		log.Errorf("failed to retrieve self-report: %v", err)
+		return nil, err
+	}
```

Comment:
Cuz you're verifying a report that was generated right before, this verification will be always successful. Then, what do you think about skipping this verification?

Reply:
This is just to retrieve `attestation.Report` struct, not for verification. The only way to get structured data is from syscall, so I could get the structured data via `enclave.VerifyRemoteReport` which returns structured `attestation.Report` data.

Do you have any idea of getting `attestation.Report` struct from byte array?

The unique ID could be given as an argument of this CLI, but I and @gyuguen thought that it's the better way to generate self-remote report and compare with it.

Reply:
The reason why I'm asking is that I'm thinking about the better way to get unique ID. It works well, but little bit weird to me as well in that generating remote report and verifying it right after, as you said.

Reply:
Ah. Gotcha. Then, I think it's good to use `VerifyRemoteReport`.